### PR TITLE
feat: Migrate to math/rand/v2

### PR DESCRIPTION
**Pull request by SoulPancake**
Solves 
- #552

### CODE REVIEW

**Comment by SoulPancake** [comment]
Please have a look @rueian  ^_^

File: internal/util/rand.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+func FastRand(n int) (r int) {
+	s := rngPool.Get().(*rand.Rand)
+	r = s.Intn(n)
+	rngPool.Put(s)
+	return
+}
+
+func BinaryString() []byte {
```
Comment (SoulPancake) [inline internal/util/rand.go:40]: On self-review, we could probably rename this 
Open to suggestions

File: internal/util/rand.1.22.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"encoding/binary"
+	"math/rand/v2"
+)
+
+func Shuffle(n int, swap func(i, j int)) {
+	rand.Shuffle(n, swap)
+}
+
+func IntN(n int) int {
```
Comment (rueian) [inline internal/util/rand.1.22.go:15]: until.IntN looks vague. I think we can just replace it with the below FastRand.

Comment (SoulPancake) [reply internal/util/rand.1.22.go:15]: We can do that,  but for go !1.22 
The regular usage of Intn would also be replaced with rngPool implementation, Should we do that? @rueian

Comment (rueian) [reply internal/util/rand.1.22.go:15]: Yes, that is okay.

File: internal/util/rand.go (hunk 1 above)
Comment (rueian) [reply internal/util/rand.go:40]: Indeed. Maybe RandomBytes.

**Comment by SoulPancake** [comment]
Addressed @rueian Pls take a look

File: om/json.go [hunk 3]
```diff
@@ -3,6 +3,7 @@ package om
 import (
 	"context"
 	"encoding/json"
+	"github.com/oklog/ulid/v2"
```
Comment (rueian) [inline om/json.go:6]: We should sort imports.

File: rueidis.go [hunk 4]
```diff
@@ -7,8 +7,8 @@ import (
 	"context"
 	"crypto/tls"
 	"errors"
+	"github.com/redis/rueidis/internal/util"
```
Comment (rueian) [inline rueidis.go:10]: We should sort imports.

File: rueidislock/lock.go [hunk 5]
```diff
@@ -2,9 +2,8 @@ package rueidislock
 
 import (
 	"context"
-	"encoding/binary"
 	"errors"
-	"math/rand"
+	"github.com/redis/rueidis/internal/util"
 	"strconv"
```
Comment (rueian) [inline rueidislock/lock.go:7]: We should sort imports.

File: sentinel.go [hunk 6]
```diff
@@ -5,15 +5,14 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"github.com/redis/rueidis/internal/util"
```
Comment (rueian) [inline sentinel.go:8]: We should sort imports.

**Review by rueian — COMMENTED** [review]
Thanks. Almost done. We just need to group and sort imports.

**Comment by SoulPancake** [comment]
@rueian Done, I guess this sorting should be okay

**Comment by rueian** [comment]
> @rueian Done, I guess this sorting should be okay

Yes, that is okay now. We just need to wait for https://github.com/redis/rueidis/issues/554.

**Comment by rueian** [comment]
> > @rueian Done, I guess this sorting should be okay
> 
> Yes, that is okay now. We just need to wait for #554.

Hi @SoulPancake, #554 has been merged. Could you rebase this PR?

**Comment by SoulPancake** [comment]
Done @rueian

**Comment by SoulPancake** [comment]
@rueian Is this ready to be merged?

**Comment by rueian** [comment]
Thanks @SoulPancake!

**Pull request merged**